### PR TITLE
Features/fres 1406 animation vote submited

### PR DESCRIPTION
**Pull request by maxired**
this PR implements several transitions when a round is finished and user is switching to a new round

### CODE REVIEW

File: elements/reigns/src/features/voting/useOnFrescoStateUpdate.ts [hunk 1]
```diff
@@ -47,3 +63,19 @@ export const useOnFrescoStateUpdate = () => {
...
+    await wait(TELEPORT_DELAY)
+    teleport('neutral');
+    await wait(RESSOURCE_DELAY)
+    dispatch(updateGame({ ...storedState.game, stats: state.stats }))
+    dispatch(startQuestionAndAnswersTransition())
+    await wait(QUESTION_CHANGE_DELAY);
+    dispatch(updateGame({ ...storedState.game, ...state }));
+    dispatch(endQuestionAndAnswersTransition())
+    dispatch(endRoundTransition())
+}
```
Comment (staff0rd) [inline elements/reigns/src/features/voting/useOnFrescoStateUpdate.ts:79]: this part is particularly great

File: elements/reigns/src/constants.ts [hunk 2]
```diff
@@ -11,3 +11,8 @@ export enum Loading {
 }
 
 export const INFINITE_CARD_WEIGHT = 100;
+
+export const CARD_FILL_ANIMATION_DURATION = 100;
+export const TELEPORT_DELAY = 100;
+export const RESSOURCE_DELAY = 400;
```
Comment (staff0rd) [inline elements/reigns/src/constants.ts:17]:
```suggestion
export const RESOURCE_DELAY = 400;
```

Comment (staff0rd) [reply elements/reigns/src/constants.ts:17]: tho probably `RESOURCE_CHANGE_DELAY` too?

File: elements/reigns/src/MeterArrow.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  currentAnimation: string;
+}) => {
+  const textRef = useRef("\u00a0"); // default value of insecapable space to have initial height
+
+  if (currentAnimation !== "") {
+    let nextText = currentAnimation.includes("--grow") ? ">" : "<";
+    if (currentAnimation.includes("--big")) {
+      nextText += nextText;
+    }
+    textRef.current = nextText; // no need to ever set back the textRef. This allows fadeout to works
```
Comment (staff0rd) [inline elements/reigns/src/MeterArrow.tsx:17]:
```suggestion
    textRef.current = nextText; // no need to ever set back the textRef. This allows fadeout to work
```

File: elements/reigns/src/constants.ts [hunk 4]
```diff
@@ -11,3 +11,8 @@ export enum Loading {
 }
 
 export const INFINITE_CARD_WEIGHT = 100;
+
+export const CARD_FILL_ANIMATION_DURATION = 100;
+export const TELEPORT_DELAY = 100;
```
Comment (staff0rd) [inline elements/reigns/src/constants.ts:16]: possibly `START_TELEPORT_DELAY`? This appears to be the 100ms we wait before teleporting, and not the ~1sec we wait for teleporting to be completed across all clients.

File: elements/reigns/src/features/voting/useRoundedRectangleProgress.tsx [hunk 5]
```diff
@@ -10,29 +11,44 @@ const ease = Quad.easeIn;
...
   const currentPositionRef = useRef<number>(0);
+  const currentFillRef = useRef<number>(0);
 
   useEffect(() => {
     if (pixi) {
       const { perimeter } = getPerimeter(pixi);
       const endPosition = perimeter * progress;
 
-      const tween = Tween.to(currentPositionRef, {
+      const positionTween = Tween.to(currentPositionRef, {
```
Comment (staff0rd) [inline elements/reigns/src/features/voting/useRoundedRectangleProgress.tsx:27]: `progressTween` and `currentProgressRef` might be clearer.

File: elements/reigns/src/features/voting/useRoundedRectangleProgress.tsx [hunk 6]
```diff
@@ -10,29 +11,44 @@ const ease = Quad.easeIn;
...
+    const fillTween = Tween.to(currentFillRef, {
+      current: fill ? 100 : 0,
+      duration: CARD_FILL_ANIMATION_DURATION / 1000,
+      ease,
+      delay: fill ? duration : 0,
+    });
+
+    return () => {
+      fillTween.kill();
+    };
```
Comment (staff0rd) [inline elements/reigns/src/features/voting/useRoundedRectangleProgress.tsx:49]: This tween and its animation is kind of weird, 100ms is super fast and the bezier shape made me think the animation was flickering on my machine. A different easing might make it less weird, ~~but I kind of think it should be zero delay, and then the whole progress (without the reverse tween) & fill fade out with the question and the answer.~~ I mean, i'm no designer :)

File: elements/reigns/src/features/voting/useRoundedRectangleProgress.tsx [hunk 7]
```diff
@@ -250,3 +267,32 @@ const getPerimeter = (app: Application) => {
...
+  current: number,
+  color: string
+) => {
+  const g = app.stage.children[0] as Graphics;
+
+  const ratio = 1.1 - (current / 100) * 1.2;
+  const height = app.screen.height;
+  const width = app.screen.width;
+
+  g.clear().lineStyle({ color: utils.string2hex(color), width: 0 });
```
Comment (staff0rd) [inline elements/reigns/src/features/voting/useRoundedRectangleProgress.tsx:282]: might be clearer (hah) to move `clear` up to the the parent so the calling function looks something like:

```
const onTick = () => {
  const g = app.stage.children[0] as Graphics;
  g.clear();
  drawFilledRectangleProgress(g, currentFillRef.current, color);
  drawRoundedRectangleProgress(g, currentPositionRef.current, color);
};
```

File: elements/reigns/src/style.css [hunk 8]
```diff
@@ -77,14 +78,39 @@ button {
...
+.meter__progress--shrink.meter__arrow,
+.meter__progress--shrink--big.meter__arrow {
+  transition: opacity 0.3s ease, color 0s;
+  color: red;
+  opacity: 1;
+}
+
+.meter__progress--grow.meter__arrow,
+.meter__progress--grow--big.meter__arrow {
+  transition: opacity 0.3s ease, color 0s;
```
Comment (staff0rd) [inline elements/reigns/src/style.css:109]: how does 0.3s fit in with that timeline above?

**Review by staff0rd — APPROVED** [review]

File: elements/reigns/src/features/voting/useRoundedRectangleProgress.tsx (hunk 6 above)
Comment (maxired) [reply elements/reigns/src/features/voting/useRoundedRectangleProgress.tsx:49]: I personnaly made this animation and like it. but agree that 100ms is too quick.
I used 100ms as required ;-)

I also agree that the fill out should be changed, maybe as a opacity change ?

File: elements/reigns/src/style.css (hunk 8 above)
Comment (maxired) [reply elements/reigns/src/style.css:109]: matching the already exiting color transition on the percent element 
https://github.com/fres-co/fresco-community/blob/d8f4642ed07ab6fe4388d0f083cf0d8f9390c1a5/elements/reigns/src/style.css#L146

will add a variable

**Pull request merged**